src/ffi.rs

```diff
@@ -356,12 +356,21 @@ pub extern fn quiche_conn_send(conn: &mut Connection, out: *mut u8,
 
 #[no_mangle]
 pub extern fn quiche_conn_stream_recv(conn: &mut Connection, stream_id: u64,
-                                      max_len: usize) -> *const RangeBuf {
-    match conn.stream_recv(stream_id, max_len) {
-        Ok(b) => Box::into_raw(Box::new(b)),
+                                      out: *mut u8, out_len: usize, fin: *mut bool)
```

Comment:
You can actually make `fin` a reference directly, .i.e.:
```rust
                                       out: *mut u8, out_len: usize, fin: &mut bool)
```
so you don't need the `.as_mut().unwrap()` later.